pkgs/http_client_conformance_tests/lib/src/redirect_tests.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:http/http.dart';
+import 'package:test/test.dart';
+
+/// Tests that the [Client] correctly implements HTTP redirect logic.
+void testRedirect(Client client) async {
+  group('redirects', () {
+    late HttpServer server;
+    setUp(() async {
+      //        URI |  Redirects TO
+      // ===========|==============
+      // ".../loop" |    ".../loop"
+      //   ".../10" |       ".../9"
+      //    ".../9" |       ".../8"
+      //        ... |           ...
+      //    ".../1" |           "/"
+      //        "/" |  <no redirect>
+      server = (await HttpServer.bind('localhost', 0))
+        ..listen((request) async {
+          if (request.requestedUri.pathSegments.isEmpty) {
+            unawaited(request.response.close());
+          } else if (request.requestedUri.pathSegments.last == 'loop') {
+            unawaited(request.response
+                .redirect(Uri.parse('http://localhost:${server.port}/loop')));
```

Comment:
[optional] I prefer using the more specific constructors since we have more semantic info than parse needs.

```suggestion
                .redirect(Uri.http('localhost:${server.port}', 'loop')));
```

Or in this case since we repeat this snippet a bunch, maybe add a little utility like

```dart
extension LocalhostPath on HttpServer {
  Uri localhostUri([String path = '']) => Uri.http('localhost:$port', path);
}
```

Then usage would be

```suggestion
                .redirect(server.localhostUri('loop')));
```



Similar throughout. If you'd find the edits annoying feel free to skip and I can handle in a followup.

Reply:
Done the first variation. I don't like having too many helpers in my tests because it reduces readability.